### PR TITLE
Remove useless tracing and useless code

### PR DESCRIPTION
**Pull request by derrickstolee**
The `OnGetFileStream()` method is a very busy area of code when hydrating a repo. Don't bog it down with useless tracing that doesn't help anything.

While exploring the code base for this excessive tracing, I found some code that can be safely deleted because it isn't used by anything.

### CODE REVIEW

File: GVFS/GVFS.Platform.Mac/MacFileSystemVirtualizer.cs [hunk 1]
```diff
@@ -355,22 +355,19 @@ private bool TryGetSymLinkTarget(string sha, out string symLinkTarget)
                 metadata.Add(nameof(sha), sha);
                 metadata.Add(nameof(placeholderVersion), placeholderVersion);
                 metadata.Add(nameof(commandId), commandId);
-                ITracer activity = this.Context.Tracer.StartActivity("GetFileStream", EventLevel.Verbose, Keywords.Telemetry, metadata);
```
Comment (jamill) [inline GVFS/GVFS.Platform.Mac/MacFileSystemVirtualizer.cs]: If this is a hot path - should we only create the event metadata if we are going to use it (i.e. inside the following if block)?

Comment (derrickstolee) [reply GVFS/GVFS.Platform.Mac/MacFileSystemVirtualizer.cs]: Good idea. I can move it to a private method to avoid code duplication, too. Thanks!

Comment (derrickstolee) [reply GVFS/GVFS.Platform.Mac/MacFileSystemVirtualizer.cs]: We can do that on the Mac, but not Windows (at least not easily), as there is an async handler that takes the existing metadata and adds to it if necessary.

**Comment by derrickstolee** [comment]
We confirmed with the WinES team that they are no longer measuring performance around this activity.

File: GVFS/GVFS.Platform.Mac/MacFileSystemVirtualizer.cs [hunk 2]
```diff
@@ -349,28 +349,20 @@ private bool TryGetSymLinkTarget(string sha, out string symLinkTarget)
                 string sha = GetShaFromContentId(contentId);
                 byte placeholderVersion = GetPlaceholderVersionFromProviderId(providerId);
 
-                EventMetadata metadata = this.CreateEventMetadata(relativePath);
```
Comment (derrickstolee) [inline GVFS/GVFS.Platform.Mac/MacFileSystemVirtualizer.cs]: We need to undo these changes to the Mac platform, as we will want to trace this data carefully while we are still learning about the platform. @jeschu1 is on it. (I'll strip out the change in this PR, though.)

File: GVFS/GVFS.Platform.Windows/WindowsFileSystemVirtualizer.cs [hunk 3]
```diff
@@ -939,10 +935,9 @@ private HResult QueryFileNameHandler(string virtualPath)
                         streamGuid,
                         sha,
                         metadata,
-                        activity),
+                        this.Context.Tracer),
```
[the remark this answers is not in the document]
Comment (derrickstolee) [reply GVFS/GVFS.Platform.Windows/WindowsFileSystemVirtualizer.cs]: Thanks, @jeschu1 for fixing this!

**Comment by derrickstolee** [comment]
Since the comment is hidden now, we are not changing `GVFS/GVFS.Platform.Mac/MacFileSystemVirtualizer.cs` as we actually want to trace that data while we are figuring out the mac platform. It is helpful to see which processes are hydrating the repo.

**Pull request merged**